### PR TITLE
feature for AppStoreServerNotification: support env type

### PR DESCRIPTION
**Pull request by oleksandr-roskovynskyi**

### CODE REVIEW

**Comment by imdhemy** [comment]
Hi @oleksandr-roskovynskyi 
Could you please use this [PR template](https://github.com/imdhemy/laravel-in-app-purchases/blob/master/.github/pull_request_template.md)? This helps me to understand the context and investigations you made!

**Comment by oleksandr-roskovynskyi** [comment]
What?

`
public function isTest(): bool    {        
    return ServerNotification::ENV_SANDBOX === $this-&gt;notification-&gt;environment;    
}`

Why?

Method isTest() of AppStoreServerNotification did not work

How?

support env type from ServerNotification (environment)

Does your code follow the PSR-2 standard? (Yes|No).

Answer: Yes

Does the psalm tool show no errors? (Yes|No).

Answer: Yes

Are there unit tests related to this PR? (Yes|No).

Answer: No

**Comment by oleksandr-roskovynskyi** [comment]
@imdhemy Please, merge

File: src/ServerNotifications/AppStoreServerNotification.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ public function getSubscription(): SubscriptionContract
      */
     public function isTest(): bool
     {
-        return false;
+        return ServerNotification::ENV_SANDBOX === $this->notification->environment;
```
Comment (imdhemy) [inline src/ServerNotifications/AppStoreServerNotification.php:49]: @oleksandr-roskovynskyi 
Did you check the [access modifier](https://github.com/oleksandr-roskovynskyi/appstore-iap/blob/main/src/ServerNotifications/ServerNotification.php#L39) of the `$environment` property?

**Comment by imdhemy** [comment]
@oleksandr-roskovynskyi 
I can understand your idea behind this PR, but I'm not sure whether this should be done this way or not. Let me clarify:

- Google Play allows sending test notifications. Thus, in the Google notification class, we can check the notification type.
- App Store does not allow sending this type of notifications. Thus, the `isTest` method returns `false` always.

Umm, What if Apple allowed sending a testing notification in the future?

**Pull request closed without merging**